database/sql/src/main/scala/cromwell/database/slick/SlickDatabase.scala

```diff
@@ -164,14 +164,10 @@ abstract class SlickDatabase(override val originalDatabaseConfig: Config) extend
   private val debugExitStatusCodeOption = ConfigFactory.load.getAs[Int](debugExitConfigPath)
 
   protected[this] def runTransaction[R](action: DBIO[R], isolationLevel: TransactionIsolation = TransactionIsolation.RepeatableRead): Future[R] = {
-    runActionInternal(action.transactionally.withTransactionIsolation(isolationLevel))
+    runAction(action.transactionally.withTransactionIsolation(isolationLevel))
   }
 
   protected[this] def runAction[R](action: DBIO[R]): Future[R] = {
-    runActionInternal(action.withPinnedSession)
```

Comment:
I don't think this file needs-to or should-be reverted. Changing this line effectively changes `runAction` to spread composite database actions across as many database connections as possible. Instead, I believe callers would expect composite commands to run in the same connection.

Reply:
I was under impression that this line is the source of the bug, as this is the code called by metadata inserts. 

I'm all for improvements but for the sake of stability I think we revert and go from there.

Reply:
Revert away, but this should be re-fixed in the future for `writeWorkflowHeartbeats`.

> I was under impression that this line is the source of the bug

Switching from `runAction` to `runTransaction` will effectively run `addMetadataEntries` in a [single session anyway, due to the call to `action.transactionally`](http://slick.lightbend.com/doc/3.3.0/dbio.html#transactions).

There are two reverts currently in this PR:
1. `addMetadataEntries` now calls `runTransaction` instead of `runAction`
2. `runAction` now calls `database.run(action)` instead of `database.run(action.withPinnedSession)`

If 1. is reverted, then `addMetadataEntries` won't be affected by changes to `runAction`. Meanwhile, reverting 2. ends up changing the `writeWorkflowHeartbeats` that still calls `runAction`, removing the session isolation.

https://github.com/broadinstitute/cromwell/blob/6e76a0e8d8e7a1c70ed18c2dff30307832866453/database/sql/src/main/scala/cromwell/database/slick/WorkflowStoreSlickDatabase.scala#L119-L120